src/to_vec.jl

```diff
@@ -6,40 +6,39 @@ transformation.
 """
 function to_vec(x::Number)
     function Number_from_vec(x_vec)
-        return first(x_vec)
+        return oftype(x, first(x_vec))
     end
     return [x], Number_from_vec
 end
 
-# Vectors
+# (Abstract)Vectors
 to_vec(x::Vector{<:Number}) = (x, identity)
```

Comment:
why is this one still `Vector` not `AbstractVector`? Can we have a comment for it?

Reply:
Good point. Let me know if the new commenting clarifies things.

Reply:
perfect! thanks :)